{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pickle\n",
    "import numpy as np\n",
    "import torch\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Returning UndirectedInteractionNetwork object.\n",
      "Continuing with Gene_A and Gene_B as columns for the nodes\n",
      "14034 Nodes and 278974 interactions\n",
      "Returning UndirectedInteractionNetwork object.\n",
      "Continuing with Gene_A and Gene_B as columns for the nodes\n",
      "1037 Nodes and 14099 interactions\n",
      "(14099, 2)\n"
     ]
    }
   ],
   "source": [
    "cancer_type = 'Neuroblastoma'\n",
    "#cancer_type = 'Non-Small Cell Lung Cancer_Lung Neuroendocrine Tumor'\n",
    "BASE_PATH = \"./Data/\"\n",
    "ppi = \"Reactome\"\n",
    "remove_rpl = \"_noRPL\"\n",
    "remove_commonE = \"\"\n",
    "useSTD = \"STD\"\n",
    "crispr_threshold_pos = -1.5\n",
    "cell_feat_name = \"cnv\"\n",
    "gene_feat_name = 'cgp'\n",
    "\n",
    "with open(BASE_PATH+f\"multigraphs/{cancer_type.replace(' ', '_')}_{ppi}{remove_rpl}_{useSTD}{remove_commonE}_crispr{str(crispr_threshold_pos).replace('.','_')}.pickle\", 'rb') as handle:\n",
    "    mg_obj = pickle.load(handle) #3columns: gene_A (gene or cell line), gene_B, type (scaffold or depmap)\n",
    "#type2nodes and int2gene are both functions defined in NetworkAnalys\n",
    "#all_genes_int is a list of integers that represent the genes \n",
    "#all_genes_name is a list of gene names\n",
    "all_genes_int = mg_obj.type2nodes['gene'] \n",
    "all_genes_name = [mg_obj.int2gene[i] for i in all_genes_int]\n",
    "\n",
    "# PPI obj\n",
    "ppi_obj = mg_obj.getEdgeType_subset(edge_type='scaffold') #gene-gene interactions\n",
    "ppi_obj_new_gene2int = {n:i for i, n in enumerate(all_genes_name)}\n",
    "ppi_obj_new_int2gene = {v:k for k, v in ppi_obj_new_gene2int.items()}\n",
    "ppi_interactions = ppi_obj.getInteractionNamed() #function defined in multigraph.py, it returns a dataframe with 2 columns \"Gene_A\", \"Gene_B\"\n",
    "ppi_interactions = ppi_interactions.map(lambda x: ppi_obj_new_gene2int[x]) #replace gene names with their corresponding integers\n",
    "\n",
    "# DEP obj\n",
    "dep_obj = mg_obj.getEdgeType_subset(edge_type='depmap')\n",
    "cells = [k for k, v in mg_obj.node_type_names.items() if v == 'cell']\n",
    "cell2int = {c:i for i, c in enumerate(cells)}\n",
    "int2cell = {v:k for k, v in cell2int.items()}\n",
    "dep_interactions = dep_obj.getInteractionNamed() #2 columns: Gene_A (=cell line) and Gene_B (=gene)\n",
    "dep_genes = [dep_obj.int2gene[i] for i in dep_obj.type2nodes['gene']] #list of genes with a dependecy edge\n",
    "\n",
    "dep_interactions.loc[~dep_interactions.Gene_A.isin(cells), ['Gene_A', 'Gene_B']] = \\\n",
    "    dep_interactions.loc[~dep_interactions.Gene_A.isin(cells), ['Gene_B', 'Gene_A']].values # assure that all values in Gene_A are cells, otherwise switch with the gene column\n",
    "\n",
    "assert dep_interactions.Gene_A.isin(cells).sum() == dep_interactions.shape[0] #all Gene_A should be cell lines\n",
    "dep_interactions = dep_interactions.map(lambda x: cell2int[x] if x in cell2int else ppi_obj_new_gene2int[x]) #map cell lines and genes to their resp. integers\n",
    "dep_interactions = dep_interactions[['Gene_B', 'Gene_A']] #switch columns (gene, cell line)\n",
    "print(dep_interactions.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "path = BASE_PATH+'Depmap/OmicsCNGene.csv'\n",
    "ccle_cnv = pd.read_csv(path, header=0, index_col=0)\n",
    "ccle_cnv.columns = [i.split(' ')[0] for i in ccle_cnv.columns] \n",
    "ccle_cnv = ccle_cnv[ccle_cnv.columns[ccle_cnv.isna().sum() == 0]] #remove columns with missing values\n",
    "ccle_cnv = ccle_cnv.loc[list(set(cells) & set(ccle_cnv.index))] #filter only the cells that are in the cell lines\n",
    "\n",
    "hvg_q = ccle_cnv.std().quantile(q=0.95)  #compute the 95th percentile of the standard deviation per gene accross all cells\n",
    "hvg_final = ccle_cnv.std()[ccle_cnv.std() >= hvg_q].index #select genes with high variance (top 5%)\n",
    "\n",
    "ccle_cnv_hvg = ccle_cnv[hvg_final]\n",
    "#cell_feat = torch.from_numpy(ccle_cnv_hvg.loc[cell2int.keys()].values).to(torch.float) #cell feature matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(37, 1218)\n",
      "             CROCCP2     CROCC  BX284668.2  BX284668.5    RNU1-2      TYW3  \\\n",
      "ACH-000149  2.325705  2.192354    2.325705    2.325705  2.325705  0.978057   \n",
      "ACH-002083  0.722578  0.722578    0.722578    0.722578  0.722578  0.722578   \n",
      "ACH-000227  0.534714  0.534714    0.534714    0.534714  0.534714  0.534714   \n",
      "ACH-001300  0.922100  0.628134    0.601304    0.601304  0.601304  0.950683   \n",
      "ACH-001603  0.509894  0.509894    0.509894    0.509894  0.509894  1.000209   \n",
      "\n",
      "                LHX8   SLC44A5     ACADM   RABGGTB  ...     GPR85    SMIM30  \\\n",
      "ACH-000149  0.978057  0.978057  0.978057  0.978057  ...  1.483138  1.483138   \n",
      "ACH-002083  0.722578  0.722578  0.722578  0.722578  ...  1.428792  1.428792   \n",
      "ACH-000227  0.534714  0.534714  0.534714  0.534714  ...  1.034856  1.034856   \n",
      "ACH-001300  0.950683  0.950683  0.950683  0.950683  ...  1.394232  1.394232   \n",
      "ACH-001603  1.000209  1.000209  1.000209  1.000209  ...  1.003859  1.003859   \n",
      "\n",
      "             PPP1R3A     FOXP2      ING3     CPED1     WNT16     FAM3C  \\\n",
      "ACH-000149  1.483138  1.483138  1.483138  1.483138  1.483138  1.483138   \n",
      "ACH-002083  1.428792  1.428792  1.067992  1.067992  1.067992  1.067992   \n",
      "ACH-000227  1.034856  1.034856  1.034856  1.034856  1.034856  1.034856   \n",
      "ACH-001300  1.394232  1.394232  1.394232  1.394232  1.394232  1.394232   \n",
      "ACH-001603  1.003859  1.003859  1.003859  1.003859  1.003859  1.003859   \n",
      "\n",
      "               PRSS1     PRSS2  \n",
      "ACH-000149  1.483138  1.483138  \n",
      "ACH-002083  1.067992  1.067992  \n",
      "ACH-000227  1.034856  1.034856  \n",
      "ACH-001300  1.394232  1.394232  \n",
      "ACH-001603  1.003859  1.003859  \n",
      "\n",
      "[5 rows x 1218 columns]\n"
     ]
    }
   ],
   "source": [
    "print(ccle_cnv_hvg.shape)\n",
    "print(ccle_cnv_hvg.head())\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dit is dus mijn cell feature matrix (CNV data). Wat ik nu moet doen is de top 100 of 200 meest variabele data eruit nemen en kijken of deze genen ook in de literatuur van neuroblastoma voorkomen."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(37, 200)\n",
      "                MYCNOS      MYCNUT        MYCN        DDX1  LINC01804  \\\n",
      "ACH-000149    1.411926    1.411926    1.411926    1.411926   1.411926   \n",
      "ACH-002083    2.010568    2.010568    2.010568    2.010568   2.010568   \n",
      "ACH-000227  146.633202  146.633202  146.633202    0.988912   0.988912   \n",
      "ACH-001300    0.946186    0.946186    0.946186    0.946186   0.946186   \n",
      "ACH-001603   32.897795   32.897795   32.897795    1.484014   1.484014   \n",
      "ACH-000203   97.644305   97.644305   97.644305   97.644305  97.644305   \n",
      "ACH-000260    1.039791    1.039791    1.039791    1.039791   1.039791   \n",
      "ACH-001301   20.681171   20.681171   20.681171   20.681171  20.681171   \n",
      "ACH-000259  118.338606  118.338606  118.338606    1.440714   1.440714   \n",
      "ACH-000310   28.482029   28.482029   28.482029   28.482029  28.482029   \n",
      "ACH-001338  132.342626  132.342626  132.342626  129.622738   0.947914   \n",
      "ACH-000312    1.497977    1.497977    1.497977    1.497977   1.497977   \n",
      "ACH-002922    1.136062    1.136062    1.136062    1.136062   1.136062   \n",
      "ACH-001303   56.024191   56.024191   56.024191   25.289113  56.024191   \n",
      "ACH-000099   15.668607   15.668607   15.668607   15.668607  15.668607   \n",
      "ACH-001366   42.027537   42.027537   42.027537    1.453125   1.453125   \n",
      "ACH-001481    1.413915    1.413915    1.413915    1.413915   1.413915   \n",
      "ACH-002282   53.472418   53.472418   53.472418   30.116550  53.472418   \n",
      "ACH-002283   41.392285   41.392285   41.392285   41.392285  41.392285   \n",
      "ACH-000366   40.032228   40.032228   40.032228   40.032228  40.032228   \n",
      "ACH-001367   18.407898   18.407898   18.407898    1.117417   1.117417   \n",
      "ACH-002284    1.242584    1.242584    1.242584    1.242584   1.242584   \n",
      "ACH-000078   58.966391   58.966391   58.966391    1.212312   1.212312   \n",
      "ACH-002261    8.038509    8.038509    8.038509    8.038509   8.038509   \n",
      "ACH-000120   35.944696   35.944696   35.944696   35.944696  35.944696   \n",
      "ACH-002278    8.575501    8.575501    8.575501    0.934989   0.934989   \n",
      "ACH-002285   48.312647   48.312647   48.312647    0.922906   0.922906   \n",
      "ACH-002080   53.124667   53.124667   53.124667   53.124667  53.124667   \n",
      "ACH-001302   84.279275   84.279275   84.279275   84.279275  84.279275   \n",
      "ACH-001354   20.510660   20.510660   20.510660    1.250675  20.510660   \n",
      "ACH-001548    7.029626    7.029626    7.029626    0.963876   7.029626   \n",
      "ACH-000804    1.164704    1.164704    1.164704    1.164704   1.164704   \n",
      "ACH-001188    1.567842    1.567842    1.567842    1.567842   1.567842   \n",
      "ACH-000341    0.983252    0.983252    0.983252    0.983252   0.983252   \n",
      "ACH-001344    1.042790    1.042790    1.042790    1.042790   1.042790   \n",
      "ACH-001674   82.930622   82.930622   82.930622    1.541625   1.541625   \n",
      "ACH-002280    1.495737    1.495737    1.495737    1.495737   1.495737   \n",
      "\n",
      "                 CYRIA        NBAS        GBP2        GBP7        GBP4  ...  \\\n",
      "ACH-000149    1.411926    1.411926    0.978057    0.978057    0.978057  ...   \n",
      "ACH-002083    2.010568    1.759529    0.722578    0.722578    0.722578  ...   \n",
      "ACH-000227    0.988799    0.988912    1.023105    1.023105    1.023105  ...   \n",
      "ACH-001300    0.946186    0.946186    0.950683    0.950683    0.950683  ...   \n",
      "ACH-001603    1.461479    1.484014    1.000209    1.000209    1.000209  ...   \n",
      "ACH-000203   23.699082    4.670152    0.997275    0.997275    0.997275  ...   \n",
      "ACH-000260    1.039791    1.039791    1.062518    1.062518    1.062518  ...   \n",
      "ACH-001301    9.950971   20.681171    1.513359    1.513359    1.513359  ...   \n",
      "ACH-000259  118.338606    1.440714    0.925573    0.925573    0.925573  ...   \n",
      "ACH-000310    1.350540   28.482029    1.369846    1.369846    1.369846  ...   \n",
      "ACH-001338    0.999818  128.112964  127.773691  127.773691  127.773691  ...   \n",
      "ACH-000312    1.575421    1.497977    0.570128    0.570128    0.570128  ...   \n",
      "ACH-002922    1.136062    1.136062    0.952644    0.952644    0.952644  ...   \n",
      "ACH-001303    1.198727    1.040872    1.472445    1.472445    1.472445  ...   \n",
      "ACH-000099   15.668607    1.087889    1.022504    1.022504    1.022504  ...   \n",
      "ACH-001366   42.027537    1.453125    0.999382    0.999382    0.999382  ...   \n",
      "ACH-001481    1.413915    1.413915    1.094113    1.094113    1.094113  ...   \n",
      "ACH-002282   53.472418    0.658059    0.664124    0.664124    0.664124  ...   \n",
      "ACH-002283    1.013835   31.616528    0.529413    0.529413    0.529413  ...   \n",
      "ACH-000366    0.993979   17.985076    0.798053    0.798053    0.798053  ...   \n",
      "ACH-001367    1.796331    1.117417    1.181837    1.181837    1.181837  ...   \n",
      "ACH-002284    0.934099   31.572602    0.993526    0.993526    0.993526  ...   \n",
      "ACH-000078    1.220117    1.212312    1.578792    1.578792    1.578792  ...   \n",
      "ACH-002261    8.038509   10.888041    0.986250    0.986250    0.986250  ...   \n",
      "ACH-000120    0.995722    0.990106    0.498448    0.498448    0.498448  ...   \n",
      "ACH-002278    2.727013    0.934989    0.905035    0.905035    0.905035  ...   \n",
      "ACH-002285    0.708666    0.764341    0.661369    0.661369    0.661369  ...   \n",
      "ACH-002080    0.945097   41.411139    1.030959    1.030959    1.030959  ...   \n",
      "ACH-001302    1.096552   33.224932    0.539973    0.539973    0.539973  ...   \n",
      "ACH-001354    2.106727    1.250675    2.277759    2.277759    2.277759  ...   \n",
      "ACH-001548    1.214311    0.963876    0.987186    0.987186    0.987186  ...   \n",
      "ACH-000804    1.164704    1.164704    1.366703    1.366703    1.366703  ...   \n",
      "ACH-001188    1.567842    1.567842    1.011170    1.011170    1.011170  ...   \n",
      "ACH-000341    0.983252    0.983252    1.096481    1.096481    1.096481  ...   \n",
      "ACH-001344    1.573914    1.042790    1.134744    1.134744    1.134744  ...   \n",
      "ACH-001674   82.930622    1.234835    1.292244    1.292244    1.292244  ...   \n",
      "ACH-002280    1.281604    1.495737    1.338426    1.338426    1.338426  ...   \n",
      "\n",
      "             PITPNM2   MIR4304  PITPNM2-AS1  MPHOSPH9  C12orf65   CDK2AP1  \\\n",
      "ACH-000149  1.003093  1.003093     1.003093  1.003093  1.003093  1.003093   \n",
      "ACH-002083  1.044298  1.044298     1.044298  1.044298  1.044298  1.044298   \n",
      "ACH-000227  0.988635  0.988635     0.988635  0.988635  0.988635  0.988635   \n",
      "ACH-001300  1.361565  1.361565     1.361565  1.361565  1.361565  1.361565   \n",
      "ACH-001603  0.990808  0.990808     0.990808  0.990808  0.990808  0.990808   \n",
      "ACH-000203  1.020793  1.020793     1.020793  1.020793  1.020793  1.020793   \n",
      "ACH-000260  1.566124  1.566124     1.566124  1.566124  1.566124  1.566124   \n",
      "ACH-001301  0.988895  0.988895     0.988895  0.988895  0.988895  0.988895   \n",
      "ACH-000259  0.950826  0.950826     0.950826  0.950826  0.950826  0.950826   \n",
      "ACH-000310  1.338154  1.338154     1.338154  1.338154  1.338154  1.338154   \n",
      "ACH-001338  0.998373  0.998373     0.998373  0.998373  0.998373  0.998373   \n",
      "ACH-000312  1.065973  1.065973     1.065973  1.065973  1.065973  1.065973   \n",
      "ACH-002922  1.433112  1.433112     1.433112  1.433112  1.433112  1.433112   \n",
      "ACH-001303  0.971388  0.971388     0.971388  0.971388  0.971388  0.971388   \n",
      "ACH-000099  1.029753  1.029753     1.029753  1.029753  1.029753  1.029753   \n",
      "ACH-001366  9.746823  9.746823     9.746823  9.746823  9.746823  9.746823   \n",
      "ACH-001481  1.323859  1.323859     1.323859  1.323859  1.323859  1.323859   \n",
      "ACH-002282  1.572744  1.572744     1.572744  1.572744  1.572744  1.572744   \n",
      "ACH-002283  1.022240  1.022240     1.022240  1.022240  1.022240  1.022240   \n",
      "ACH-000366  1.201032  1.201032     1.201032  1.201032  1.201032  1.201032   \n",
      "ACH-001367  1.140474  1.140474     1.140474  1.140474  1.140474  1.140474   \n",
      "ACH-002284  0.973993  0.973993     0.973993  0.973993  0.973993  0.973993   \n",
      "ACH-000078  0.947292  0.947292     0.947292  0.947292  0.947292  0.947292   \n",
      "ACH-002261  1.173092  1.173092     1.173092  1.173092  1.173092  1.173092   \n",
      "ACH-000120  0.995583  0.995583     0.995583  0.995583  0.995583  0.995583   \n",
      "ACH-002278  0.914471  0.914471     0.914471  0.914471  0.914471  0.914471   \n",
      "ACH-002285  0.961055  0.961055     0.961055  0.961055  0.961055  0.961055   \n",
      "ACH-002080  1.044917  1.044917     1.044917  1.044917  1.044917  1.044917   \n",
      "ACH-001302  1.035068  1.035068     1.035068  1.035068  1.035068  1.035068   \n",
      "ACH-001354  0.963604  0.963604     0.963604  0.963604  0.963604  0.963604   \n",
      "ACH-001548  0.961495  0.961495     0.961495  0.961495  0.961495  0.961495   \n",
      "ACH-000804  1.124136  1.124136     1.124136  1.124136  1.124136  1.124136   \n",
      "ACH-001188  1.007677  1.007677     1.007677  1.007677  1.007677  1.007677   \n",
      "ACH-000341  1.247055  1.247055     1.247055  1.247055  1.247055  1.247055   \n",
      "ACH-001344  1.096006  1.096006     1.096006  1.096006  1.096006  1.096006   \n",
      "ACH-001674  1.218979  1.218979     1.218979  1.218979  1.218979  1.218979   \n",
      "ACH-002280  1.227298  1.227298     1.227298  1.227298  1.227298  1.227298   \n",
      "\n",
      "               SBNO1   MIR8072    RILPL2    LRRC43  \n",
      "ACH-000149  1.003093  1.003093  1.003093  1.003093  \n",
      "ACH-002083  1.044298  1.044298  1.046876  1.044298  \n",
      "ACH-000227  0.988635  0.988635  0.988635  0.988635  \n",
      "ACH-001300  1.361565  1.361565  1.361565  1.361565  \n",
      "ACH-001603  0.990808  0.990808  0.990808  0.990808  \n",
      "ACH-000203  1.020793  1.020793  1.020793  1.020793  \n",
      "ACH-000260  1.566124  1.566124  1.566124  1.566124  \n",
      "ACH-001301  0.988895  0.988895  0.988895  0.988895  \n",
      "ACH-000259  0.950826  0.950826  0.950826  0.950826  \n",
      "ACH-000310  1.338154  1.338154  1.338154  1.338154  \n",
      "ACH-001338  0.998373  0.998373  0.998373  0.998373  \n",
      "ACH-000312  1.065973  1.065973  1.065973  1.065973  \n",
      "ACH-002922  1.433112  1.433112  1.433112  1.433112  \n",
      "ACH-001303  0.971388  0.971388  0.971388  0.971388  \n",
      "ACH-000099  1.029753  1.029753  1.029753  1.029753  \n",
      "ACH-001366  9.746823  9.746823  9.746823  9.746823  \n",
      "ACH-001481  1.323859  1.323859  1.323859  1.323859  \n",
      "ACH-002282  1.572744  1.572744  1.572744  1.572744  \n",
      "ACH-002283  1.022240  1.022240  1.022240  1.022240  \n",
      "ACH-000366  1.201032  1.201032  1.201032  1.216399  \n",
      "ACH-001367  1.140474  1.140474  1.140474  1.140474  \n",
      "ACH-002284  0.973993  0.973993  0.973993  0.973993  \n",
      "ACH-000078  0.947292  0.947292  0.947292  0.947292  \n",
      "ACH-002261  1.173092  1.173092  1.173092  1.173092  \n",
      "ACH-000120  0.995583  0.995583  0.995583  0.995583  \n",
      "ACH-002278  0.914471  0.914471  0.914471  0.914471  \n",
      "ACH-002285  0.961055  0.961055  0.961055  0.961055  \n",
      "ACH-002080  1.044917  1.044917  1.044917  1.044917  \n",
      "ACH-001302  1.035068  1.035068  1.035068  1.035068  \n",
      "ACH-001354  0.963604  0.963604  0.963604  0.963604  \n",
      "ACH-001548  0.961495  0.961495  0.961495  0.961495  \n",
      "ACH-000804  1.124136  1.124136  1.124136  1.124136  \n",
      "ACH-001188  1.007677  1.007677  1.007677  1.007677  \n",
      "ACH-000341  1.247055  1.247055  1.247055  1.247055  \n",
      "ACH-001344  1.096006  1.096006  1.096006  1.096006  \n",
      "ACH-001674  1.218979  1.218979  1.218979  1.218979  \n",
      "ACH-002280  1.227298  1.227298  1.227298  1.227298  \n",
      "\n",
      "[37 rows x 200 columns]\n"
     ]
    }
   ],
   "source": [
    "#Select the 200 columns with highest variance in ccle_cnv_hvg\n",
    "cell_feat = ccle_cnv_hvg.loc[:, ccle_cnv_hvg.var().nlargest(200).index]\n",
    " \n",
    "print(cell_feat.shape)\n",
    "print(cell_feat)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['MYCNOS', 'MYCNUT', 'MYCN', 'DDX1', 'LINC01804', 'CYRIA', 'NBAS',\n",
      "       'GBP2', 'GBP7', 'GBP4',\n",
      "       ...\n",
      "       'PITPNM2', 'MIR4304', 'PITPNM2-AS1', 'MPHOSPH9', 'C12orf65', 'CDK2AP1',\n",
      "       'SBNO1', 'MIR8072', 'RILPL2', 'LRRC43'],\n",
      "      dtype='object', length=200)\n",
      "200\n"
     ]
    }
   ],
   "source": [
    "# extract column names (out of ccle_cnv_hvg)\n",
    "gene_names = cell_feat.columns\n",
    "print(gene_names)\n",
    "print(len(gene_names))\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['MYCNOS',\n",
       " 'MYCNUT',\n",
       " 'MYCN',\n",
       " 'DDX1',\n",
       " 'LINC01804',\n",
       " 'CYRIA',\n",
       " 'NBAS',\n",
       " 'GBP2',\n",
       " 'GBP7',\n",
       " 'GBP4',\n",
       " 'AC083805.1',\n",
       " 'ATP23',\n",
       " 'LINC02403',\n",
       " 'LINC02375',\n",
       " 'LINC02411',\n",
       " 'LINC02393',\n",
       " 'LINC00507',\n",
       " 'LINC02441',\n",
       " 'LINC02368',\n",
       " 'MIR3612',\n",
       " 'CTDSP2',\n",
       " 'LYZ',\n",
       " 'YEATS4',\n",
       " 'CNOT2',\n",
       " 'SLC35E3',\n",
       " 'MDM2',\n",
       " 'CPM',\n",
       " 'AC020637.1',\n",
       " 'LINC02388',\n",
       " 'TMEM132C',\n",
       " 'AVIL',\n",
       " 'MIR26A2',\n",
       " 'TSFM',\n",
       " 'MARCHF9',\n",
       " 'CYP27B1',\n",
       " 'METTL1',\n",
       " 'EEF1AKMT3',\n",
       " 'CDK4',\n",
       " 'AGAP2-AS1',\n",
       " 'TSPAN31',\n",
       " 'MIR6759',\n",
       " 'AGAP2',\n",
       " 'CPSF6',\n",
       " 'MIR1279',\n",
       " 'FRS2',\n",
       " 'MIR3913-1',\n",
       " 'MIR3913-2',\n",
       " 'C1GALT1C1L',\n",
       " 'PRANCR',\n",
       " 'CCT2',\n",
       " 'TRIB2',\n",
       " 'MIR3125',\n",
       " 'MIR4262',\n",
       " 'MIR3681',\n",
       " 'LINC01799',\n",
       " 'LINC01628',\n",
       " 'LINC01828',\n",
       " 'LINC01829',\n",
       " 'ETAA1',\n",
       " 'LINC02831',\n",
       " 'LINC00276',\n",
       " 'LRATD1',\n",
       " 'MYRFL',\n",
       " 'RAB3IP',\n",
       " 'PTPRB',\n",
       " 'KCNMB4',\n",
       " 'LRRC10',\n",
       " 'BEST3',\n",
       " 'AC025263.1',\n",
       " 'NUP107',\n",
       " 'TSPAN8',\n",
       " 'PTPRR',\n",
       " 'NTSR2',\n",
       " 'AC106875.1',\n",
       " 'MIR3681HG',\n",
       " 'LPIN1',\n",
       " 'MIR548S',\n",
       " 'PLEKHH2',\n",
       " 'RAP1B',\n",
       " 'SNORA70G',\n",
       " 'AC090061.1',\n",
       " 'CAMKMT',\n",
       " 'LINC00958',\n",
       " 'RASSF10-DT',\n",
       " 'RASSF10',\n",
       " 'ARNTL',\n",
       " 'BTBD10',\n",
       " 'PTH',\n",
       " 'FAR1',\n",
       " 'LINC02545',\n",
       " 'LINC02683',\n",
       " 'SPON1',\n",
       " 'RRAS2',\n",
       " 'OS9',\n",
       " 'MEIS1',\n",
       " 'GKN2',\n",
       " 'GKN1',\n",
       " 'MIR3126',\n",
       " 'AC064875.1',\n",
       " 'ALK',\n",
       " 'GREB1',\n",
       " 'LINC02405',\n",
       " 'YPEL5',\n",
       " 'SNORA10B',\n",
       " 'LBH',\n",
       " 'LCLAT1',\n",
       " 'ANTXR1',\n",
       " 'COPB1',\n",
       " 'OR4C11',\n",
       " 'OR4P4',\n",
       " 'OR4S2',\n",
       " 'OR4C6',\n",
       " 'WNT16',\n",
       " 'FAM3C',\n",
       " 'RAD51AP2',\n",
       " 'CCL3L3',\n",
       " 'CCL4L2',\n",
       " 'ODC1',\n",
       " 'SNORA80B',\n",
       " 'TEAD1',\n",
       " 'HPCAL1',\n",
       " 'VSNL1',\n",
       " 'RDH14',\n",
       " 'NT5C1B-RDH14',\n",
       " 'NT5C1B',\n",
       " 'LRIG3',\n",
       " 'SLC16A7',\n",
       " 'ACACB',\n",
       " 'FOXN4',\n",
       " 'THADA',\n",
       " 'CLIP4',\n",
       " 'NRBP1',\n",
       " 'KRTCAP3',\n",
       " 'IFT172',\n",
       " 'FNDC4',\n",
       " 'GCKR',\n",
       " 'C2orf16',\n",
       " 'ZNF512',\n",
       " 'CCDC121',\n",
       " 'GPN1',\n",
       " 'SUPT7L',\n",
       " 'SLC4A1AP',\n",
       " 'LINC01460',\n",
       " 'RBKS',\n",
       " 'AP000439.2',\n",
       " 'LINC02747',\n",
       " 'CCND1',\n",
       " 'LTO1',\n",
       " 'FGF19',\n",
       " 'FGF4',\n",
       " 'FGF3',\n",
       " 'LINC02753',\n",
       " 'ANO1',\n",
       " 'MYEOV',\n",
       " 'ROCK2',\n",
       " 'LINC00570',\n",
       " 'E2F6',\n",
       " 'C2orf50',\n",
       " 'SLC66A3',\n",
       " 'LINC01250',\n",
       " 'GAL',\n",
       " 'TESMIN',\n",
       " 'CPT1A',\n",
       " 'LINC02701',\n",
       " 'MRPL21',\n",
       " 'IGHMBP2',\n",
       " 'MRGPRD',\n",
       " 'MRGPRF',\n",
       " 'MRGPRF-AS1',\n",
       " 'PPP6R3',\n",
       " 'KMT5A',\n",
       " 'P2RX7',\n",
       " 'P2RX4',\n",
       " 'B3GNT4',\n",
       " 'DIABLO',\n",
       " 'VPS33A',\n",
       " 'CLIP1',\n",
       " 'ZCCHC8',\n",
       " 'RSRC2',\n",
       " 'KNTC1',\n",
       " 'HCAR2',\n",
       " 'HCAR3',\n",
       " 'HCAR1',\n",
       " 'DENR',\n",
       " 'CCDC62',\n",
       " 'HIP1R',\n",
       " 'VPS37B',\n",
       " 'ABCB9',\n",
       " 'OGFOD2',\n",
       " 'ARL6IP4',\n",
       " 'PITPNM2',\n",
       " 'MIR4304',\n",
       " 'PITPNM2-AS1',\n",
       " 'MPHOSPH9',\n",
       " 'C12orf65',\n",
       " 'CDK2AP1',\n",
       " 'SBNO1',\n",
       " 'MIR8072',\n",
       " 'RILPL2',\n",
       " 'LRRC43']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(gene_names)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Nu is de volgende stap opzoeken of deze genen effectief voorkomen in de literatuur van neuroblastoma."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Chat-GPTs answer: look up for confirmation in literature!!!!\n",
    "\n",
    "Genes Associated with Neuroblastoma\n",
    "From the provided list, here are genes that are commonly associated with neuroblastoma (NB), based on their established roles in this disease:\n",
    "\n",
    "MYCN:\n",
    "Primary oncogene in neuroblastoma. MYCN amplification is a hallmark of high-risk neuroblastoma and is used as a clinical biomarker to stratify patient risk.\n",
    "\n",
    "DDX1:\n",
    "Located close to MYCN on chromosome 2p. Often co-amplified with MYCN in neuroblastoma and thought to be involved in tumor progression.\n",
    "\n",
    "ALK:\n",
    "A major oncogene in neuroblastoma, frequently mutated in familial and sporadic neuroblastoma. ALK amplifications or point mutations are implicated in NB pathogenesis.\n",
    "\n",
    "NBAS:\n",
    "Located near MYCN; while less characterized in neuroblastoma, its proximity to MYCN suggests possible involvement in cases of co-amplification.\n",
    "\n",
    "MDM2:\n",
    "A regulator of p53, which can play a role in tumorigenesis. Overexpression or amplification of MDM2 is implicated in neuroblastoma pathogenesis, particularly in MYCN-amplified tumors.\n",
    "\n",
    "CDK4:\n",
    "Frequently amplified in neuroblastoma, particularly in the context of MYCN amplification. Promotes cell cycle progression and tumor growth.\n",
    "\n",
    "CNOT2:\n",
    "A gene linked to transcriptional regulation, including MYCN-driven pathways in neuroblastoma. However, its role is less directly studied.\n",
    "\n",
    "Why Do Other Genes Have High CNV Variability?\n",
    "Copy number variability (CNV) reflects structural alterations in the genome, such as amplifications or deletions. High CNV variability across these genes may arise due to the following reasons:\n",
    "\n",
    "1. Proximity to Known Amplification Hotspots\n",
    "Many of the listed genes are located in genomic regions known to undergo frequent amplifications or deletions in neuroblastoma or other cancers.\n",
    "Example: Genes like LINC01804, DDX1, and NBAS are in the 2p24 region, near MYCN, a hotspot for amplification in neuroblastoma.\n",
    "2. Chromosomal Instability\n",
    "Neuroblastoma is characterized by chromosomal instability, leading to frequent gains/losses in specific regions:\n",
    "2p amplification: MYCN, DDX1, NBAS.\n",
    "1p deletion: Genes such as CNOT2 and others.\n",
    "11q deletion: A common feature in non-MYCN-amplified neuroblastoma.\n",
    "3. Role in Tumorigenesis\n",
    "Even if not directly linked to neuroblastoma, genes with high variability might:\n",
    "Act as drivers (genes that promote tumor growth when altered).\n",
    "Be part of the tumor's adaptation to stress, such as hypoxia or therapy resistance.\n",
    "4. Passenger Alterations\n",
    "Some high-variability genes may simply be \"passenger\" alterations. These occur due to proximity to amplified driver genes like MYCN but do not directly contribute to tumorigenesis.\n",
    "5. Regulatory Non-Coding RNAs\n",
    "Genes like LINC02403 and LINC00507 are long non-coding RNAs (lncRNAs). Variability in these genes may alter transcriptional or epigenetic landscapes, indirectly influencing cancer progression.\n",
    "6. Role in Cellular Pathways Commonly Dysregulated in Neuroblastoma\n",
    "Genes like METTL1, CPSF6, and CDK4 participate in pathways (e.g., cell cycle, DNA damage repair, or metabolism) that are frequently dysregulated in neuroblastoma.\n",
    "7. Technical Artifacts\n",
    "CNV variability in some genes may stem from limitations in detecting CNVs accurately or regions prone to sequencing bias.\n",
    "Summary\n",
    "Neuroblastoma-associated genes: MYCN, ALK, CDK4, NBAS, DDX1, MDM2.\n",
    "High CNV variability in the other genes could be attributed to chromosomal instability, proximity to neuroblastoma hotspots, functional relevance in tumorigenic pathways, or passenger alterations. Further analysis (e.g., gene expression or functional studies) would clarify their specific role in neuroblastoma progression.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "PHD van Jonathan Linusgabre also mentions: \n",
    "MYCN, ALK, MDM2, CDK4, ODC1\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Expression data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = BASE_PATH+'Depmap/OmicsExpressionProteinCodingGenesTPMLogp1.csv'\n",
    "ccle_expression = pd.read_csv(path, header=0, index_col=0)\n",
    "ccle_expression.columns = [i.split(' ')[0] for i in ccle_expression.columns]\n",
    "# subset_nodes = list(set(ccle_expression.columns) & set(all_genes_name))\n",
    "cancer_expression = ccle_expression.loc[list(set(cells) & set(ccle_expression.index))]\n",
    "\n",
    "hvg_q = cancer_expression.std().quantile(q=0.95) #threshold for high variance genes\n",
    "hvg_final = cancer_expression.std()[cancer_expression.std() >= hvg_q].index #select genes with high variance\n",
    "\n",
    "cancer_expression_hvg = cancer_expression[hvg_final]\n",
    "# cancer_expression_full = pd.concat([cancer_expression,\n",
    "#                                     pd.DataFrame(np.tile(cancer_expression.mean().values, (len(set(cells) - set(cancer_expression.index)), 1)),\n",
    "#                                                  index=list(set(cells) - set(cancer_expression.index)), columns=cancer_expression.columns)])\n",
    "\n",
    "#filling the missing entries with the mean expression values of the corresponding genes.\n",
    "cancer_expression_full = pd.concat([cancer_expression_hvg,\n",
    "                                    pd.DataFrame(np.tile(cancer_expression_hvg.mean().values, (len(set(cells) - set(cancer_expression_hvg.index)), 1)),\n",
    "                                                index=list(set(cells) - set(cancer_expression_hvg.index)), columns=cancer_expression_hvg.columns)])\n",
    "#cancer_expression_full[:] = np.random.uniform(0, 10, size=cancer_expression_full.shape) #delete this line!!! It generates random values (between 0 and 10) for each cell \n",
    "#cell_feat = torch.from_numpy(cancer_expression_full.loc[cell2int.keys()].values).to(torch.float) #cell featture matrix with rows as cell lines and columns as genes with high variance\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(37, 960)\n",
      "              TSPAN6     FUCA2      CD99  TMEM176A      ICA1     CFLAR  \\\n",
      "ACH-000149  3.776104  6.132166  7.518220  0.545968  5.266412  4.526695   \n",
      "ACH-000227  4.033863  0.575312  6.310340  0.201634  8.303507  3.388190   \n",
      "ACH-001300  3.346248  3.695994  6.394548  0.713696  8.165661  1.941106   \n",
      "ACH-001603  3.152183  1.389567  5.432291  2.185867  7.311703  2.536053   \n",
      "ACH-000203  4.364572  3.124328  7.132577  2.189034  8.049141  3.735522   \n",
      "\n",
      "                TFPI     HSPB6     ITGA3     CRLF1  ...    CT45A1    FBXO17  \\\n",
      "ACH-000149  1.839960  0.815575  8.900474  4.678635  ...  0.367371  5.208673   \n",
      "ACH-000227  0.014355  0.344828  0.871844  2.666757  ...  0.000000  5.394034   \n",
      "ACH-001300  0.056584  0.505891  1.389567  1.416840  ...  0.000000  4.481557   \n",
      "ACH-001603  0.042644  0.042644  2.397803  2.604071  ...  0.000000  4.355439   \n",
      "ACH-000203  1.150560  1.137504  6.448901  3.917432  ...  0.000000  4.452859   \n",
      "\n",
      "             CT45A10      CD24    ZNF595     DOC2B     H4C12      CBSL  \\\n",
      "ACH-000149  0.084064  6.146085  6.678635  1.803227  3.934517  4.464668   \n",
      "ACH-000227  0.000000  9.222336  7.805615  3.240314  3.030336  5.772677   \n",
      "ACH-001300  0.000000  8.139398  8.160426  4.621759  3.615887  0.298658   \n",
      "ACH-001603  0.000000  7.880747  7.407098  3.583760  4.187451  1.150560   \n",
      "ACH-000203  0.189034  9.336261  7.437128  4.324091  2.370164  2.316146   \n",
      "\n",
      "              GAGE2E      NEFL  \n",
      "ACH-000149  0.000000  4.713696  \n",
      "ACH-000227  0.000000  5.669310  \n",
      "ACH-001300  0.000000  7.292598  \n",
      "ACH-001603  0.000000  5.603478  \n",
      "ACH-000203  0.176323  4.723012  \n",
      "\n",
      "[5 rows x 960 columns]\n"
     ]
    }
   ],
   "source": [
    "print(cancer_expression_full.shape)\n",
    "print(cancer_expression_full.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(37, 200)\n"
     ]
    }
   ],
   "source": [
    "# select the 200 columns with highest variance in cancer_expression_full\n",
    "cell_feat = cancer_expression_full.loc[:, cancer_expression_full.var().nlargest(200).index]\n",
    "print(cell_feat.shape)\n",
    "#names of the genes with high variance\n",
    "gene_names = cell_feat.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['RPS4Y1',\n",
       " 'MYCN',\n",
       " 'S100A6',\n",
       " 'PAGE5',\n",
       " 'DLK1',\n",
       " 'COL3A1',\n",
       " 'XAGE1A',\n",
       " 'S100A11',\n",
       " 'STMN2',\n",
       " 'ARHGAP36',\n",
       " 'MAGEA4',\n",
       " 'TGFBI',\n",
       " 'ANXA1',\n",
       " 'SST',\n",
       " 'COL1A1',\n",
       " 'NPY',\n",
       " 'ANXA2',\n",
       " 'S100A10',\n",
       " 'SERPINE1',\n",
       " 'IGF2',\n",
       " 'CHGA',\n",
       " 'VIP',\n",
       " 'NNAT',\n",
       " 'S100A16',\n",
       " 'RBP1',\n",
       " 'GTSF1',\n",
       " 'DBH',\n",
       " 'FABP5',\n",
       " 'NSG2',\n",
       " 'FOS',\n",
       " 'MSN',\n",
       " 'C7',\n",
       " 'CRH',\n",
       " 'CD99',\n",
       " 'LUM',\n",
       " 'RGS5',\n",
       " 'LGALS1',\n",
       " 'RGS4',\n",
       " 'CHGB',\n",
       " 'CTAG2',\n",
       " 'GDAP1L1',\n",
       " 'MAGEA10',\n",
       " 'MAGEA3',\n",
       " 'KRT18',\n",
       " 'MAGEA6',\n",
       " 'FN1',\n",
       " 'EMP3',\n",
       " 'DDX1',\n",
       " 'GAGE2A',\n",
       " 'DKK1',\n",
       " 'CYGB',\n",
       " 'IFI16',\n",
       " 'GAL',\n",
       " 'S100A4',\n",
       " 'CAVIN1',\n",
       " 'DDC',\n",
       " 'ENPP2',\n",
       " 'DDX3Y',\n",
       " 'RAB34',\n",
       " 'MAGEA12',\n",
       " 'TH',\n",
       " 'TFAP2B',\n",
       " 'CCDC80',\n",
       " 'HGF',\n",
       " 'THBS1',\n",
       " 'GFRA2',\n",
       " 'PRAME',\n",
       " 'IL13RA2',\n",
       " 'TMEM176B',\n",
       " 'CSAG1',\n",
       " 'CD44',\n",
       " 'IFITM3',\n",
       " 'CRABP1',\n",
       " 'EIF1AY',\n",
       " 'CRABP2',\n",
       " 'UBB',\n",
       " 'TMEM176A',\n",
       " 'RTN1',\n",
       " 'ASS1',\n",
       " 'RET',\n",
       " 'ITGA3',\n",
       " 'CYRIA',\n",
       " 'SLITRK6',\n",
       " 'MMP14',\n",
       " 'XAGE1B',\n",
       " 'BGN',\n",
       " 'STMN4',\n",
       " 'CYB561',\n",
       " 'NT5E',\n",
       " 'DCN',\n",
       " 'KCNQ2',\n",
       " 'HLA-C',\n",
       " 'PRSS3',\n",
       " 'MT2A',\n",
       " 'HTRA1',\n",
       " 'PRRX1',\n",
       " 'TGM2',\n",
       " 'CPLX2',\n",
       " 'GPR50',\n",
       " 'GCHFR',\n",
       " 'EGR1',\n",
       " 'TUBA1C',\n",
       " 'NEUROD1',\n",
       " 'COL4A2',\n",
       " 'CXCL12',\n",
       " 'IGFBP3',\n",
       " 'MAGEA2',\n",
       " 'MGP',\n",
       " 'YBX3',\n",
       " 'ELN',\n",
       " 'INSM2',\n",
       " 'SERPINH1',\n",
       " 'TM4SF1',\n",
       " 'ASCL1',\n",
       " 'LRRC17',\n",
       " 'TFPI2',\n",
       " 'MAGEA9B',\n",
       " 'LOXL2',\n",
       " 'ETV4',\n",
       " 'CSAG3',\n",
       " 'GSTM1',\n",
       " 'MARCHF11',\n",
       " 'ITGB5',\n",
       " 'COTL1',\n",
       " 'IER3',\n",
       " 'SCG3',\n",
       " 'EPB41L3',\n",
       " 'SPOCK1',\n",
       " 'NID1',\n",
       " 'SCG5',\n",
       " 'KIRREL1',\n",
       " 'CDKN1A',\n",
       " 'VEGFD',\n",
       " 'NEFM',\n",
       " 'TIMP3',\n",
       " 'PHOX2B',\n",
       " 'MYC',\n",
       " 'CALCB',\n",
       " 'TSPYL5',\n",
       " 'GPX7',\n",
       " 'GNG3',\n",
       " 'PRPH',\n",
       " 'BMP7',\n",
       " 'IGSF1',\n",
       " 'ATP1A3',\n",
       " 'SEMA3C',\n",
       " 'CD9',\n",
       " 'TSPAN8',\n",
       " 'HMGA2',\n",
       " 'P2RX3',\n",
       " 'EDIL3',\n",
       " 'IFI44',\n",
       " 'CDCA7L',\n",
       " 'FXYD5',\n",
       " 'RELN',\n",
       " 'SLC18A1',\n",
       " 'IGF2BP1',\n",
       " 'GPC3',\n",
       " 'PLAT',\n",
       " 'TPM2',\n",
       " 'KIF1A',\n",
       " 'PTX3',\n",
       " 'TSTD1',\n",
       " 'CBSL',\n",
       " 'LAMP5',\n",
       " 'FOSB',\n",
       " 'HPCA',\n",
       " 'LGALS3',\n",
       " 'TEAD2',\n",
       " 'CNTNAP2',\n",
       " 'NPC2',\n",
       " 'CRIM1',\n",
       " 'NQO1',\n",
       " 'SLC38A5',\n",
       " 'NDN',\n",
       " 'CSRP1',\n",
       " 'TNFRSF12A',\n",
       " 'EPAS1',\n",
       " 'FUCA2',\n",
       " 'IFITM2',\n",
       " 'GREM1',\n",
       " 'MAB21L2',\n",
       " 'PTN',\n",
       " 'CXCR4',\n",
       " 'KCNG1',\n",
       " 'H2AJ',\n",
       " 'CDH11',\n",
       " 'KRBOX1',\n",
       " 'TNC',\n",
       " 'VGF',\n",
       " 'CD24',\n",
       " 'NRN1',\n",
       " 'MFAP4',\n",
       " 'ELAVL3',\n",
       " 'CYBA',\n",
       " 'SERPINB7',\n",
       " 'MEST',\n",
       " 'B4GALNT4',\n",
       " 'CCN1',\n",
       " 'GAGE2E']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(gene_names)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Geen enkele can deze komt overeen met wat in Jonathan zijn PhD staat!!"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ChatGPT:\n",
    "\n",
    "Genes Highly Associated with Neuroblastoma\n",
    "MYCN:\n",
    "\n",
    "Key Role: MYCN amplification is a hallmark of high-risk neuroblastoma and is strongly associated with poor prognosis.\n",
    "Mechanism: MYCN drives oncogenesis by promoting cell proliferation and suppressing differentiation.\n",
    "PHOX2B:\n",
    "\n",
    "Key Role: Mutations in PHOX2B are linked to familial cases of neuroblastoma.\n",
    "Mechanism: It is crucial for the development of the autonomic nervous system and neural crest-derived tissues.\n",
    "ASCL1:\n",
    "\n",
    "Key Role: Plays a role in neuroblast differentiation and is highly expressed in neuroblastoma cells.\n",
    "Mechanism: Functions as a transcription factor in neural lineage commitment.\n",
    "CXCR4:\n",
    "\n",
    "Key Role: Overexpressed in neuroblastoma and associated with metastasis.\n",
    "Mechanism: CXCR4 mediates chemotaxis and cell migration, critical for metastatic spread.\n",
    "RET:\n",
    "\n",
    "Key Role: Activating mutations in RET are implicated in neuroblastoma and other neuroendocrine tumors.\n",
    "Mechanism: RET signaling promotes cell survival and proliferation.\n",
    "TH (Tyrosine Hydroxylase):\n",
    "\n",
    "Key Role: TH is a marker of neural crest-derived tumors, including neuroblastoma.\n",
    "Mechanism: Catalyzes dopamine synthesis, which is a characteristic feature of neuroblastoma cells.\n",
    "CHGA (Chromogranin A):\n",
    "\n",
    "Key Role: A neuroendocrine marker often elevated in neuroblastoma.\n",
    "SST (Somatostatin):\n",
    "\n",
    "Key Role: Somatostatin expression is observed in some neuroblastoma subtypes and serves as a diagnostic marker.\n",
    "TGFBI:\n",
    "\n",
    "Key Role: Promotes extracellular matrix remodeling and metastasis in aggressive neuroblastomas.\n",
    "IGF2:\n",
    "\n",
    "Key Role: The IGF2 pathway is implicated in neuroblastoma proliferation and survival."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
